src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/jpa/entity/LicenceConditionChangeAuditEntity.kt

```diff
@@ -1,31 +1,52 @@
 package uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.entity
 
-import jakarta.persistence.CascadeType
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import jakarta.persistence.AttributeConverter
 import jakarta.persistence.Column
+import jakarta.persistence.Convert
+import jakarta.persistence.Converter
 import jakarta.persistence.Entity
 import jakarta.persistence.GeneratedValue
 import jakarta.persistence.GenerationType
 import jakarta.persistence.Id
-import jakarta.persistence.JoinColumn
-import jakarta.persistence.OneToOne
 import jakarta.persistence.Table
+import org.hibernate.annotations.JdbcTypeCode
+import org.hibernate.type.SqlTypes
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.LicenceConditions
 import java.time.LocalDateTime
 
 @Entity
 @Table(name = "licence_conditions_change_audit")
 data class LicenceConditionChangeAuditEntity(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
-  val id: Long?,
+  val id: Long? = null,
 
-  @OneToOne(cascade = [CascadeType.MERGE])
-  @JoinColumn(name = "prisoner_id", referencedColumnName = "id")
-  val prisoner: PrisonerEntity,
+  @Column
+  val prisonerId: Long,
+  @Column
+  val version: Int = 1,
```

Comment:
do you want to add @Version ? for pessimistic locking

Reply:
Sure, although you can only set it to true at the moment so the updates should be idempotent regardless